modules/l4tls/handler.go

```diff
@@ -104,6 +120,209 @@ func (t *Handler) Handle(cx *layer4.Connection, next layer4.Handler) error {
 	return next.Handle(cx.Wrap(tlsConn))
 }
 
+// FIXME: Should this perhaps be moved instead to caddytls?
```

Comment:
Let's start by trying it here first, then it'll be easier to consider moving it upstream. :+1: